app/components/listboxes.tsx

```diff
@@ -35,7 +35,7 @@ export function HorseListbox({
 	const [selected, setSelected] = useState(initialValues)
 
 	return (
-		<Listbox value={selected} onChange={setSelected} name={name} multiple>
+		<Listbox value={selected} by="id" onChange={setSelected} name={name} multiple>
```

Comment:
Adding by="id" fixes a bug that existed where if the form has an error, the selected horses are not able to be deselected, and can be double selected, etc.